controllers/disruption_controller.go

```diff
@@ -182,6 +182,37 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			}
 
 			return ctrl.Result{Requeue: true}, err
+		} else if calculateRemainingDuration(*instance) <= 0 {
+			if _, err := r.updateInjectionStatus(instance); err != nil {
+				r.log.Errorw("error updating disruption injection status", "error", err)
+
+				return ctrl.Result{}, fmt.Errorf("error updating disruption injection status: %w", err)
+			}
+
+			isCleaned, err := r.cleanDisruption(instance)
```

Comment:
I see that we call clean disruption here, how were we handling the cleanup before that (on expiration)?

Reply:
We were letting the chaos pods clean themselves up (they had a TTL set). This works most of the time (well it works all of the time if the chaos pods injected correctly), but it's helpful to clean explicitly here just to be safe but also to handle chaos pods that got stuck